src/st_widgets.c

```diff
@@ -645,16 +651,23 @@ void ST_ResetTitle(void)
                '0' + hudcolor_titl, string);
 
     announce_string[0] = '\0';
+    author_string[0] = '\0';
     if (hud_map_announce && leveltime == 0)
     {
         if (gamemapinfo && gamemapinfo->author)
         {
             M_snprintf(announce_string, sizeof(announce_string), "%s by %s",
                        string, gamemapinfo->author);
+            if (MN_StringWidth(announce_string) > SCREENWIDTH) 
```

Comment:
Should we really only act if the string spans the whole screen?

Reply:
> Should we really only act if the string spans the whole screen?

It only spans the entire screen with 4:3 aspect ratio, `#define SCREENWIDTH 320`. I think it works fine with 16:9, which is what most users have.